File: app/src/main/java/com/bitsanddroids/myplants/userauthentication/RegisterActivity.java
package com.bitsanddroids.myplants.userauthentication;

import android.nfc.Tag;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Toast;

import com.bitsanddroids.myplants.R;
import com.google.android.gms.tasks.OnCompleteListener;
import com.google.android.gms.tasks.OnFailureListener;
import com.google.android.gms.tasks.OnSuccessListener;
import com.google.android.gms.tasks.Task;
import com.google.firebase.FirebaseApp;
import com.google.firebase.auth.AuthResult;
import com.google.firebase.auth.FirebaseAuth;
import com.google.firebase.auth.FirebaseUser;
import com.google.firebase.auth.UserProfileChangeRequest;
import com.google.firebase.firestore.DocumentReference;
import com.google.firebase.firestore.FirebaseFirestore;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

public class RegisterActivity extends AppCompatActivity {

    private FirebaseAuth auth;

    private EditText usernameEditText;
    private EditText emailEditText;
    private EditText passwordEditText;

    private Button registerButton;



    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.register_user_page);

        usernameEditText = findViewById(R.id.usernameEditText);
        emailEditText = findViewById(R.id.emailEditText);
        passwordEditText = findViewById(R.id.passwordEditText);

        registerButton = findViewById(R.id.registerButton);
        FirebaseApp.initializeApp(this);
        auth = FirebaseAuth.getInstance();

        registerButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                createAccount();
            }
        });
    }


    private void createAccount(){
        String email = emailEditText.getText().toString();
        String password = passwordEditText.getText().toString();
        auth.createUserWithEmailAndPassword(email, password).addOnCompleteListener(this, new OnCompleteListener<AuthResult>() {
            @Override
            public void onComplete(@NonNull Task<AuthResult> task) {
                if(task.isSuccessful()){
                    final FirebaseUser userDB = auth.getCurrentUser();
                    //check if user is really pressent
                    if (userDB != null) {
                        //create user object to store personalplants and username
                        User user = new User(userDB.getUid(), usernameEditText.getText().toString());
                        FirebaseFirestore db = FirebaseFirestore.getInstance();
                        db.collection("users")
                                .document(userDB.getUid())
                                .set(user);
                    }
                } else {
                    Toast.makeText(RegisterActivity.this, "Could not register this account", Toast.LENGTH_SHORT).show();
                }
            }
        });

    }
}

File: app/src/main/java/com/bitsanddroids/myplants/userauthentication/RegisterActivity.kt
package com.bitsanddroids.myplants.userauthentication

import android.os.Bundle
import android.widget.Button
import android.widget.EditText
import android.widget.Toast

import com.bitsanddroids.myplants.R
import com.google.firebase.FirebaseApp
import com.google.firebase.auth.FirebaseAuth
import com.google.firebase.firestore.FirebaseFirestore
import androidx.appcompat.app.AppCompatActivity

class RegisterActivity : AppCompatActivity() {

    private var auth: FirebaseAuth? = null

    private var usernameEditText: EditText? = null
    private var emailEditText: EditText? = null
    private var passwordEditText: EditText? = null

    private var registerButton: Button? = null


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.register_user_page)

        usernameEditText = findViewById(R.id.usernameEditText)
        emailEditText = findViewById(R.id.emailEditText)
        passwordEditText = findViewById(R.id.passwordEditText)

        registerButton = findViewById(R.id.registerButton)
        FirebaseApp.initializeApp(this)
        auth = FirebaseAuth.getInstance()

        registerButton!!.setOnClickListener { createAccount() }
    }


    private fun createAccount() {
        val email = emailEditText!!.text.toString()
        val password = passwordEditText!!.text.toString()
        auth!!.createUserWithEmailAndPassword(email, password).addOnCompleteListener(this) { task ->
            if (task.isSuccessful) {
                val userDB = auth!!.currentUser
                //check if user is really pressent
                if (userDB != null) {
                    //create user object to store personalplants and username
                    val user = User(userDB.uid, usernameEditText!!.text.toString(), null,1)
                    val db = FirebaseFirestore.getInstance()
                    db.collection("users")
                            .document(userDB.uid)
                            .set(user)
                }
            } else {
                Toast.makeText(this@RegisterActivity, "Could not register this account", Toast.LENGTH_SHORT).show()
            }
        }

    }
}
